src/dataverse_client/rest_client.py

```diff
@@ -0,0 +1,205 @@
+"""Client for interacting with the Dataverse API"""
+
+from typing import Optional
+
+import msal
+from pydantic import SecretStr, computed_field
+from pydantic_settings import BaseSettings
+import requests
+
+
+class DataverseConfig(
+    BaseSettings,
+    validate_default=True,
+    env_prefix="DATAVERSE_",
+):
+    """
+    Configuration settings for the Dataverse client.
+    Loads from environment variables prefixed with "DATAVERSE_"
+    """
+
+    tenant_id: str = "32669cd6-737f-4b39-8bdd-d6951120d3fc"
+    client_id: str = "df37356e-3316-484a-b732-319b6b4ad464"
+    org: str = "orgc1997c24"
+
+    additional_scopes: list[str] = ["offline_access"]
+
+    username: str = "svc_sipe"
+    password: SecretStr
+
+    @computed_field
+    @property
+    def api_url(self) -> str:
+        """Base URL for the Dataverse API."""
+        return f"https://{self.org}.crm.dynamics.com/api/data/v9.2/"
+
+    @computed_field
+    @property
+    def env_url(self) -> str:
+        """Base URL for the Dataverse environment."""
+        return f"https://{self.org}.crm.dynamics.com"
+
+    @computed_field
+    @property
+    def authority(self) -> str:
+        """Base URL for the Azure AD authority."""
+        return f"https://login.microsoftonline.com/{self.tenant_id}"
+
+    @computed_field
+    @property
+    def scope(self) -> str:
+        """Scope for the Dataverse API."""
+        return f"{self.env_url}/.default " + " ".join(self.additional_scopes)
+
+
+class DataverseRestClient:
+    """Client for basic CRUD operations on Dataverse entities."""
+
+    def __init__(self, config: DataverseConfig):
+        """
+        Initialize the DataverseRestClient with configuration.
+        Acquires an authentication token and sets up request headers.
+        Args:
+            config (DataverseConfig): Config object with credentials and URLs.
+        """
+        self.config = config
+        self.token = self._acquire_token()
+        self.headers = {
+            "Authorization": f"Bearer {self.token['access_token']}",
+            "OData-MaxVersion": "4.0",
+            "OData-Version": "4.0",
+            "Accept": "application/json",
+            "If-None-Match": None,
+            "Content-Type": "application/json",
+        }
+
+    def _acquire_token(self):
+        """
+        Acquire an access token using MSAL and user credentials.
+        Returns:
+            dict: Token dictionary containing 'access_token'.
+        Raises:
+            ValueError: If token acquisition fails.
+        """
+        app = msal.PublicClientApplication(
+            client_id=self.config.client_id,
+            authority=self.config.authority,
+            client_credential=None,
+        )
+
+        if not self.config.username.endswith("@alleninstitute.org"):
```

Comment:
if we're going to do this autocomplete (I don't think it's necessary, but I don't care that much), let's put this in a settings file somewhere.